### PR TITLE
build: rename std.Build.*Step to std.Build.Step.*

### LINKED ISSUE #14947: rename std.Build.FooStep to std.Build.Step.Foo
Extracted from #14647.

### PR DESCRIPTION
**Pull request by sterchelen**
Follow-up actions from #14647

Fixes #14947

### CODE REVIEW

File: lib/std/Build/Step/CheckFile.zig [hunk 1]
```diff
@@ -2,6 +2,11 @@
 //! TODO: make this more flexible, supporting more kinds of checks.
 //! TODO: generalize the code in std.testing.expectEqualStrings and make this
 //! CheckFileStep produce those helpful diagnostics when there is not a match.
+const CheckFileStep = @This();
+const std = @import("std");
+const Step = std.Build.Step;
+const fs = std.fs;
+const mem = std.mem;
```
Comment (sterchelen) [inline lib/std/Build/Step/CheckFile.zig:9]: Saw `const` at bottom or at the top of the file... 
There are more `const` at the top for Steps files so I took advantage of doing this PR to make things more consistent.

File: lib/std/Build/Step/CheckObject.zig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const std = @import("../std.zig");
+const std = @import("std");
```
Comment (marler8997) [inline lib/std/Build/Step/CheckObject.zig:1]: Interesting.  Should files in the `std` module always be importing std by alias like this now?  If so what's the reason(s)?  I'm curious if non-std modules should be adopting the same convention.

Comment (leecannon) [reply lib/std/Build/Step/CheckObject.zig:1]: Just to note in order to do this type of thing in non-std modules you have to add the module to its own dependencies: https://github.com/ziglang/zig/issues/14708#issuecomment-1440770273

Comment (sterchelen) [reply lib/std/Build/Step/CheckObject.zig:1]: I was wondering if we could avoid relative path on std import. It seems yes, we are already doing this in `lib/std/Build/Cache.zig` https://github.com/ziglang/zig/blob/5df31f3ef3d2c4640fa2c7fc9b03c83c6a8606ae/lib/std/Build/Cache.zig#L76

**Comment by sterchelen** [comment]
@andrewrk do you want me to resolve conflicts before reviewing that PR?

**Review by Vexu — APPROVED** [review]

File: lib/std/Build.zig [hunk 3]
```diff
@@ -29,20 +29,20 @@ pub const Builder = Build;
...
+pub const InstallArtifactStep = @import("Build/Step/InstallArtifact.zig");
+pub const InstallDirStep = @import("Build/Step/InstallDir.zig");
+pub const InstallFileStep = @import("Build/Step/InstallFile.zig");
+pub const ObjCopyStep = @import("Build/Step/ObjCopy.zig");
+pub const CompileStep = @import("Build/Step/Compile.zig");
+pub const OptionsStep = @import("Build/Step/Options.zig");
+pub const RemoveDirStep = @import("Build/Step/RemoveDir.zig");
+pub const RunStep = @import("Build/Step/Run.zig");
+pub const TranslateCStep = @import("Build/Step/TranslateC.zig");
+pub const WriteFileStep = @import("Build/Step/WriteFile.zig");
```
Comment (nektro) [inline lib/std/Build.zig:45]: these decls are in the right file location but not the right namespace

**Pull request merged**